src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -126,8 +126,23 @@ void CreateSpaceProcessor::process(const cpp2::CreateSpaceReq& req) {
             return;
         }
 
+        auto hostLoadingRet = getHostLoading();
+        if (!hostLoadingRet.ok()) {
+            LOG(ERROR) << "Get host loading failed";
+            handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
+            onFinished();
+            return;
+        }
+
+        auto hostLoading = hostLoadingRet.value();
         for (auto partId = 1; partId <= partitionNum; partId++) {
-            auto partHosts = pickHostsWithZone(partId, zones, replicaFactor);
+            auto partHostsRet = pickHostsWithZone(partId, zones, replicaFactor, hostLoading);
```

Comment:
Do we need to check size of zones is greater than replicaFactor?

Reply:
yep  I have check it in Line 122